multy_core/src/ethereum/ethereum_stream.cpp

```diff
@@ -0,0 +1,122 @@
+/* Copyright 2018 by Multy.io
+ * Licensed under Multy.io license.
+ *
+ * See LICENSE for details
+ */
+
+#include "multy_core/src/ethereum/ethereum_stream.h"
+
+#include "multy_core/src/ethereum/ethereum_extra_data.h"
+
+#include "multy_core/src/exception_stream.h"
+#include "multy_core/src/utility.h"
+#include "multy_core/binary_data.h"
+
+#include <array>
+#include <unordered_map>
+#include <vector>
+
+namespace
+{
+using namespace multy_core::internal;
+
+const size_t ETHEREUM_SIZE_VARIABLE_FUNCTION_CONTRACT = 32;
+static const std::array<const uint8_t, ETHEREUM_SIZE_VARIABLE_FUNCTION_CONTRACT> ZEROES = {0};
+
+} // namespace
+
+
+namespace multy_core
+{
+namespace internal
+{
+
+EthereumPayloadDataStream::EthereumPayloadDataStream()
+        :m_data()
+{
+    m_data.reserve(256);
+}
+
+EthereumPayloadDataStream::~EthereumPayloadDataStream()
+{
+}
+
+BinaryDataPtr EthereumPayloadDataStream::get_content() const
+{
+    return make_clone(as_binary_data(m_data));
+}
+
+void EthereumPayloadDataStream::write_data(const void* data, size_t size)
+{
+    INVARIANT(data != nullptr);
+    const uint8_t* d = reinterpret_cast<const uint8_t*>(data);
+    m_data.insert(m_data.end(), d, d + size);
+}
+
+EthereumPayloadAsUint256Stream::EthereumPayloadAsUint256Stream()
+        :m_data()
+{
+    m_data.reserve(256);
+}
+
+EthereumPayloadAsUint256Stream::~EthereumPayloadAsUint256Stream()
+{
+}
+
+BinaryDataPtr EthereumPayloadAsUint256Stream::get_content() const
```

Comment:
Please make that function return plain `BinaryData`, like every other data stream does.